math/check_amicable_pair.cpp

```diff
@@ -0,0 +1,102 @@
+/**
+ * Copyright 2020 @author iamnambiar
+ * 
+ * @file
+ * A C++ Program to check whether a pair of number is amicable pair or not.
+ * 
+ * \brief
+ * Amicable Pair are two different numbers such that sum of the proper divisor 
+ * of each number is equal to the other number.
+ *  
+ * Wikipedia link for Reference: https://en.wikipedia.org/wiki/Amicable_numbers 
+ */
+#include <iostream>
+#include <cassert>
+#include <math.h>
+
+/**
+ * Function to calculate the sum of all the proper divisor 
+ * of an integer.
+ * @param num First number.
+ * @return Sum of the proper divisor of the number.
+ */
+int sum_of_divisor(int num) {
+    // Variable to store the sum of all proper divisors.
+    int sum = 0;
+    for(int div = 2; div <= sqrt(num); ++div) {
```

Comment:
Specify why its being running till sqrt(num), so that people who don't know this optimization  can understand it

Reply:
I have added comments in the updated one.